staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,289 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberc
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/config/v1alpha1"
+	"k8s.io/kubectl/pkg/scheme"
+)
+
+const RecommendedKubeRCFileName = ".kuberc"
+
+var (
+	RecommendedConfigDir  = filepath.Join(homedir.HomeDir(), clientcmd.RecommendedHomeDir)
+	RecommendedKubeRCFile = filepath.Join(RecommendedConfigDir, RecommendedKubeRCFileName)
+)
+
+// PreferencesHandler is responsible for setting default flags
+// arguments based on user's kuberc configuration.
+type PreferencesHandler interface {
+	AddFlags(flags *pflag.FlagSet)
+	Apply(rootCmd *cobra.Command, args []string, errOut io.Writer) ([]string, error)
+}
+
+// Preferences stores the kuberc file coming either from environment variable
+// or file from set in flag or the default kuberc path.
+type Preferences struct {
+	GetPreferencesFunc func(kuberc string) (*v1alpha1.Preferences, error)
+
+	aliases map[string]struct{}
+}
+
+// NewPreferences returns initialized Prefrences object.
+func NewPreferences() PreferencesHandler {
+	return &Preferences{
+		GetPreferencesFunc: DefaultGetPreferences,
```

Comment:
Why this is needed at all? I'd exepct `getPreferences` to just be a function of the `ProcessHandler`

Reply:
Unit tests are passing their own customized `GetPreferencesFunc`. But we can unexport it, if you'd want.

Reply:
Yeah, let's make it private for now. If we decide in the long run we need, we can always export it. But if people start relying on that API we'll be forced to support it.

Reply:
Updated.